SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -0,0 +1,164 @@
+}
+
+# --------------------------------------------------------------------------------------------
+
+# Parse command line arguments
+while [ $# -gt 0 ]; do
+    key="$1"
+    case $key in
+        --env|-e)
+            ENV_NAME=$2
+            shift
+            shift
+            ;;
+        --repo|-r)
+            REPOSITORY=$2
+            shift
+            shift
+            ;;
+        --accpy)
+            ACCPY_VERSION=$2
+            shift
+            shift
+            ;;
+        --help|-h)
+            print_help
+            exit 0
+            ;;
+        *)
+            >&2 _log "ERROR: Invalid argument: $1" && _log
+            print_help
+            exit 1
+            ;;
+    esac
+done
+
+# --------------------------------------------------------------------------------------------
+
+# Checks if a name for the environment is given
+if [ -z "$ENV_NAME" ]; then
+    >&2 _log "ERROR: No virtual environment name provided." && _log
+    print_help
+    exit 1
+fi
+
+# Checks if the provided Acc-Py version is valid
+if [ -n "$ACCPY_VERSION" ] && [ ! -e "$ACCPY_PATH/base/$ACCPY_VERSION" ]; then
+    >&2 _log "ERROR: Invalid Acc-Py version."
+    exit 1
+fi
+
+# Checks if a repository file is given
+if [ -z "$REPOSITORY" ]; then
+    >&2 _log "ERROR: No repository provided." && _log
+    print_help
+    exit 1
+# Checks if the provided repository source is found
+elif [ -d $REPOSITORY ]; then
+    REQ_PATH="${REPOSITORY}/requirements.txt"
+    if [ ! -f "${REQ_PATH}" ]; then
+        >&2 _log "ERROR: Requirements file not found (${REQ_PATH})."
+        exit 1
+    fi
+elif [[ $REPOSITORY == http* ]]; then
+    # Extract the repository name from the URL
+    repo_name=$(basename $REPOSITORY)
+    repo_name=${repo_name%.*}
+    
+    mkdir -p $HOME/SWAN_projects
+
+    define_repo_path $repo_name
+
+    # Clone the repository
+    echo "Cloning the repository from ${REPOSITORY}..."
+    git clone $REPOSITORY -q "${REPO_PATH}" || { >&2 _log "ERROR: Failed to clone repository"; exit 1; }
+
+    REQ_PATH=${REPO_PATH}/requirements.txt
+    # Check if requirements.txt exists in the repository
+    if [ ! -f "${REQ_PATH}" ]; then
+        rm -rf ${REPO_PATH}
+        >&2 _log "ERROR: ${REQ_PATH} not found in ${REPO_PATH}."
+        exit 1
+    fi
+else
+    >&2 _log "ERROR: Invalid repository (${REPOSITORY})."
+    exit 1
+fi
+
+
+# --------------------------------------------------------------------------------------------
+
+ENV_PATH="/home/$USER/${ENV_NAME}"
+
+# Create virtual environment (acc-py or generic)
+if [ -n "$ACCPY_VERSION" ]; then
+    source $ACCPY_PATH/base/${ACCPY_VERSION}/setup.sh
+    if [ -d "${ENV_PATH}" ]; then
+        rm -rf ${ENV_PATH}
+    fi
+    acc-py venv ${ENV_PATH}
+else
+    message="Creating"
+    [ -d "${ENV_PATH}" ] && message="Recreating"
+    echo "${message} virtual environment ${ENV_NAME} using Generic Python..."
+    
+    python -m venv ${ENV_PATH} --clear --copies
```

Comment:
You add the `--clear` because under some circumstances we might be recreating the environment (e.g. running this script twice)? Is it when the script fails to run and you give the user the option to retry?

Reply:
*answered above

Reply:
So we do not need to specify here the clear option anymore after our discussion today?

Reply:
Clear is not needed anymore.